tests/testsuite/sbom.rs

```diff
@@ -0,0 +1,333 @@
+}
+
+#[cargo_test]
+fn build_sbom_with_simple_build_script() {
+    let p = configured_project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.0.1"
+                authors = []
+                build = "build.rs"
+            "#,
+        )
+        .file("src/main.rs", "#[cfg(foo)] fn main() {}")
+        .file(
+            "build.rs",
+            r#"fn main() { println!("cargo::rustc-cfg=foo"); }"#,
+        )
+        .build();
+
+    p.cargo("build -Zsbom")
+        .masquerade_as_nightly_cargo(&["sbom"])
+        .run();
+
+    let path = p.bin("foo").with_extension("cargo-sbom.json");
+    assert!(path.is_file());
+}
+
+#[cargo_test]
+fn build_sbom_with_build_dependencies() {
+    Package::new("baz", "0.1.0").publish();
+    Package::new("bar", "0.1.0")
+        .build_dep("baz", "0.1.0")
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "bar"
+                version = "0.1.0"
+                build = "build.rs"
+
+                [build-dependencies]
+                baz = "0.1.0"
+            "#,
+        )
+        .file("src/lib.rs", "pub fn bar() -> i32 { 2 }")
+        .file(
+            "build.rs",
+            r#"fn main() { println!("cargo::rustc-cfg=foo"); }"#,
+        )
+        .publish();
+
+    let p = configured_project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.0.1"
+                authors = []
+
+                [dependencies]
+                bar = "0.1.0"
+            "#,
+        )
+        .file("src/main.rs", "fn main() { let _i = bar::bar(); }")
+        .build();
+
+    p.cargo("build -Zsbom")
+        .masquerade_as_nightly_cargo(&["sbom"])
+        .run();
+
+    let path = p.bin("foo").with_extension("cargo-sbom.json");
+    assert_json_output(
+        path,
+        r#"
+        {
+            "format_version": 1,
+            "package_id": "path+file:///[..]/foo#0.0.1",
+            "name": "foo",
+            "version": "0.0.1",
+            "source": "[ROOT]/foo",
+            "target": {
+                "kind": [
+                    "bin"
+                ],
+                "crate_type": "bin",
+                "name": "foo",
+                "edition": "2015"
+            },
+            "profile": {
+                "name": "dev",
+                "opt_level": "0",
+                "lto": "false",
+                "codegen_backend": null,
+                "codegen_units": null,
+                "debuginfo": 2,
+                "split_debuginfo": "{...}",
+                "debug_assertions": true,
+                "overflow_checks": true,
+                "rpath": false,
+                "incremental": false,
+                "panic": "unwind",
+                "strip": {
+                    "deferred": "None"
+                }
```

Comment:
This needs to be resolved. Deferred is not too meaningful for SBOM if we do want this field.

Reply:
It's the output of [`Profile`](https://github.com/rust-lang/cargo/blob/master/src/cargo/core/profiles.rs#L50) struct, but I think it's more sensible to not expose this type directly, in case its properties ever change or new fields are added.

Reply:
The `strip` field got removed from the JSON output.